packages/react/src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,153 @@
+import * as customPropTypes from '@stardust-ui/react-proptypes'
+import * as _ from 'lodash'
+import * as React from 'react'
+import * as PropTypes from 'prop-types'
+
+import {
+  applyAccessibilityKeyHandlers,
+  AutoControlledComponent,
+  createShorthandFactory,
+  ChildrenComponentProps,
+  commonPropTypes,
+  isFromKeyboard,
+  UIComponentProps,
+} from '../../lib'
+import { ComponentEventHandler, WithAsProp, ShorthandValue, withSafeTypeForAs } from '../../types'
+import Icon from '../Icon/Icon'
+import Text from '../Text/Text'
+import { Accessibility } from '../../lib/accessibility/types'
+import { checkboxBehavior } from '../../lib/accessibility'
+
+export interface CheckboxProps extends UIComponentProps, ChildrenComponentProps {
```

Comment:
Let's add conformat test for the new component.

Reply:
Added, have to add an additional handler for `onChange`. Decided to keep it flat to be more obvious